pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -149,80 +149,38 @@ func (r *RepoSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
-		return controllerruntime.Result{}, updateErr
-	}
-
-	if errs := validation.IsDNS1123Subdomain(reconcilerRef.Name); errs != nil {
-		err := errors.Errorf("Invalid reconciler name %q: %s.", reconcilerRef.Name, strings.Join(errs, ", "))
-		r.logger(ctx).Error(err, "Sync name or namespace invalid")
-		reposync.SetStalled(rs, "Validation", err)
-		// Validation errors should not trigger retry (return error),
-		// unless the status update also fails.
-		_, updateErr := r.updateStatus(ctx, currentRS, rs)
-		// Use the validation error for metric tagging.
-		metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
-		return controllerruntime.Result{}, updateErr
-	}
-
-	if err := r.validateSpec(ctx, rs, reconcilerRef.Name); err != nil {
-		r.logger(ctx).Error(err, "Sync spec invalid")
-		reposync.SetStalled(rs, "Validation", err)
-		// Validation errors should not trigger retry (return error),
-		// unless the status update also fails.
-		_, updateErr := r.updateStatus(ctx, currentRS, rs)
-		// Use the validation error for metric tagging.
-		metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
-		return controllerruntime.Result{}, updateErr
-	}
-
-	if err := r.validateValuesFileSourcesRefs(ctx, rs); err != nil {
-		r.logger(ctx).Error(err, "Sync spec invalid")
-		reposync.SetStalled(rs, "Validation", err)
-		// Validation errors should not trigger retry (return error),
-		// unless the status update also fails.
-		_, updateErr := r.updateStatus(ctx, currentRS, rs)
-		// Use the validation error for metric tagging.
-		metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
 
-		// We want to allow deletion of ConfigMap and RSync to occur in any order.
-		// For example, if a namespace is deleted, the ConfigMap and RSync can
-		// be deleted in a random order. That means that validation errors
-		// from the ConfigMap should halt reconciliation progress only if the RSync
-		// is not being deleted. If the RSync is being deleted, we still need to finalize
-		// the RSync and run the teardown function.
-		if rs.GetDeletionTimestamp().IsZero() {
+		// Validate the RepoSync after the ConfigMap watch is set up, so new
+		// ConfigMaps can be validated.
+		if err := r.validateRepoSync(ctx, rs, reconcilerRef.Name); err != nil {
+			r.logger(ctx).Error(err, "Invalid RepoSync Spec")
+			reposync.SetStalled(rs, "Validation", err)
+			// Validation errors should not trigger retry (return error),
+			// unless the status update also fails.
+			_, updateErr := r.updateStatus(ctx, currentRS, rs)
+			// Use the validation error for metric tagging.
+			metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
 			return controllerruntime.Result{}, updateErr
 		}
-		if updateErr != nil {
-			r.logger(ctx).Error(updateErr, "Failed to update sync status")
-		}
+	} else {
+		// Deletion requested.
+		// Cleanup is handled above, after the RepoSync is NotFound.
```

Comment:
this comment line is no longer accurate and can be removed. yes, we do delete above if not found, but ideally it's already cleaned up by the teardown below.

Reply:
done